### PR TITLE
#166841012 Integrate social Login

### PR DESCRIPTION
**Pull request by Alpha1202**
#### What does this PR do?
Integrate Social Login for google, facebook and twitter

#### Description of Task to be completed?
 - install passport
- install passport-facebook
- install passport-google-oauth20
- install passport-twitter
- initialise passport strategies

Have the following endpoints working

`GET /api/auth/google`
`GET /api/auth/facebook`
`GET /api/auth/twitter`

#### How should this be manually tested?
After cloning the repo, cd into the folder and RUN `npm run start:dev`

- on your choice browser, go to any of the routes above

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [#166841012](https://www.pivotaltracker.com/story/show/166841012)

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/A

### CODE REVIEW

**Review by phembarl — APPROVED** [review]

**Review by codeBlock-1984 — APPROVED** [review]
Great job @Alpha1202. Looks good to go

**Review by czarjulius — APPROVED** [review]
Looks good to me

File: src/config/social_config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+const googleStrategy = new GoogleStrategy(googleConfig, socialCallBack);
+
+const facebookStrategy = new FacebookStrategy(facebookConfig, socialCallBack);
+
+const twitterStrategy = new TwitterStrategy(twitterConfig, socialCallBack);
+
+const gitHubStrategy = new GitHubStrategy(githubConfig, socialCallBack);
+
+const SerializeSetUp = () => {
```
Comment (fob413) [inline src/config/social_config.js:22]: please document this function (jsdocs)

File: src/controllers/socialController.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      const userData = req.authInfo;
+      if (userData.emails === undefined) {
+        return res.status(400).send('Email not found');
+      }
+      const newPassword = Math.random().toString(36).slice(-10);
+      const password = hashPassword(newPassword);
+
+
+      let firstName, lastName;
+      let lName;
```
Comment (fob413) [inline src/controllers/socialController.js:28]: Line 27 and 28, can be just a single line

File: src/controllers/socialController.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          password,
+          isVerified: false,
+          bio: ' ',
+          verificationToken: ' ',
+          email: userData.emails[0].value,
+          imageUrl: userData.photos[0].value,
+          social: userData.provider
+        }
+      });
+
```
Comment (fob413) [inline src/controllers/socialController.js:53]: Please remove all white spaces

**Review by fob413 — CHANGES_REQUESTED** [review]
Please make these quick changes

**Pull request closed without merging**